packages/cli/src/ExternalSecrets/providers/azure-key-vault/azure-key-vault.ts

```diff
@@ -0,0 +1,127 @@
+import { ClientSecretCredential } from '@azure/identity';
+import { SecretClient } from '@azure/keyvault-secrets';
+import type { SecretsProvider, SecretsProviderState } from '@/Interfaces';
+import type { INodeProperties } from 'n8n-workflow';
+import type { AzureKeyVaultContext, AzureKeyVaultSecret } from './types';
+import { DOCS_HELP_NOTICE, EXTERNAL_SECRETS_NAME_REGEX } from '@/ExternalSecrets/constants';
+
+export class AzureKeyVault implements SecretsProvider {
+	name = 'azureKeyVault';
+
+	displayName = 'Azure Key Vault';
+
+	state: SecretsProviderState = 'initializing';
+
+	properties: INodeProperties[] = [
```

Comment:
I don't know why we decided to reuse nodes code for external-secrets and for log-streaming, but I really hope we can migrate away from this pattern soon.